src/Core/ExternalTable.cpp

```diff
@@ -55,23 +70,61 @@ void BaseExternalTable::clear()
 
 void BaseExternalTable::parseStructureFromStructureField(const std::string & argument)
 {
-    std::vector<std::string> vals;
-    splitInto<' ', ','>(vals, argument, true);
+    /// First try to parse table structure with `ParserNameTypePairList`, this allows user to declare Enum types in the structure
+    ParserNameTypePairList parser;
+    const auto * pos = argument.data();
+    String error;
+    ASTPtr columns_list_raw = tryParseQuery(parser, pos, pos+argument.size(), error, false, "", false, 0, 0);
+    bool parse_structure_with_parser = false;
+    if ((parse_structure_with_parser = columns_list_raw != nullptr))
+    {
+        for (auto & child : columns_list_raw->children)
+        {
+            auto * column = child->as<ASTNameTypePair>();
+            if (column)
+                structure.emplace_back(column->name, column->type->getColumnNameWithoutAlias());
+            else
+            {
+                structure.clear();
+                parse_structure_with_parser = false;
+                break;
+            }
+        }
+    }
+
+    if (!parse_structure_with_parser)
```

Comment:
I see this fallback as unneeded. The previous code was simply wrong. Splitting by comma is always the wrong way to parse something. Your code is better.